### ISSUE TITLE
Error while executing zstd - error on save

### ISSUE THREAD

**Issue by mmajewsk** [hlissner/doom-emacs#3788]
**What did you expect to happen?**
Just to save a file/buffer

**What actually happened?**
File was saved but there was additional message in half bottom of the screen

**Additional details:**
the additional message
```
Error while executing "zstd -c -q < /tmp/jka-comV3F5nO"

/bin/sh: 1: zstd: not found
```

I have enabled debug but there is no backtrace.

**Steps to reproduce:**
1. Open doom
2. SPC q L  to load session
3. Edit file
4. :w to write

**System information:**
<details><pre>

```
emacs   version    26.3
        features   XPM JPEG TIFF GIF PNG RSVG IMAGEMAGICK SOUND GPM DBUS GSETTINGS GLIB NOTIFY LIBSELINUX GNUTLS LIBXML2 FREETYPE M17N_FLT LIBOTF XFT ZLIB TOOLKIT_SCROLL_BARS GTK3 X11 XDBE XIM MODULES THREADS XWIDGETS LIBSYSTEMD LCMS2
        build      Sep 16, 2019
        buildopts  (--build=x86_64-linux-gnu --prefix=/usr '--includedir=${prefix}/include' '--mandir=${prefix}/share/man' '--infodir=${prefix}/share/info' --sysconfdir=/etc --localstatedir=/var --disable-silent-rules '--libdir=${prefix}/lib/x86_64-linux-gnu' '--libexecdir=${prefix}/lib/x86_64-linux-gnu' --disable-maintainer-mode --disable-dependency-tracking --prefix=/usr --sharedstatedir=/var/lib --program-suffix=26 --with-modules --with-file-notification=inotify --with-mailutils --with-x=yes --with-x-toolkit=gtk3 --with-xwidgets --with-lcms2 'CFLAGS=-g -O2 -fdebug-prefix-map=/build/emacs26-TP6iDo/emacs26-26.3~1.git96dd019=. -fstack-protector-strong -Wformat -Werror=format-security -no-pie' 'CPPFLAGS=-Wdate-time -D_FORTIFY_SOURCE=2' 'LDFLAGS=-Wl,-Bsymbolic-functions -Wl,-z,relro -no-pie')
        windowsys  x
        daemonp    server-running
doom    version    2.0.9
        build      grafted, HEAD -> develop, origin/develop, origin/HEAD 9c1a356 2020-07-28 20:08:32 -0400
        dir        ~/.doom.d/
system  type       gnu/linux
        config     x86_64-pc-linux-gnu
        shell      /bin/bash
        uname      Linux 5.3.0-40-generic #32~18.04.1-Ubuntu SMP Mon Feb 3 14:05:59 UTC 2020 x86_64
        path       (/snap/direnv/54/usr/sbin/ /snap/direnv/54/usr/bin/ /snap/direnv/54/sbin/ /snap/direnv/54/bin/ /usr/local/sbin/ /usr/local/bin/ /usr/sbin/ /usr/bin/ /sbin/ /bin/ /usr/games/ /usr/local/games/ ~/repositories/lartpc/lartpc2D-rl/ /usr/lib/x86_64-linux-gnu/emacs/26.3/x86_64-linux-gnu/)
config  envfile    envvar-file
        elc-files  0
        modules    (:completion company ivy :ui doom doom-dashboard doom-quit hl-todo modeline nav-flash neotree ophints (popup +defaults) vc-gutter vi-tilde-fringe window-select workspaces :editor (evil +everywhere) file-templates fold multiple-cursors snippets :emacs dired electric undo vc :checkers syntax :tools direnv docker ein (eval +overlay) lookup magit make :lang cc data emacs-lisp latex markdown org sh :config (default +bindings +smartparens))
        packages   ((cmake-ide) (elpy) (conda))
        unpin      (n/a)
        elpa       (n/a)
```
</pre></details>

**Comment by hlissner** [comment]
The `:emacs undo` module has [this advice](https://github.com/hlissner/doom-emacs/blob/develop/modules/emacs/undo/config.el#L42-L49). It causes undo-fu-session to save undo-fu history compressed _if_ `zstd` is found in the system at startup.

According to the error, `zstd` is not on your system, and yet it is still trying to use it. The only way this can happen is if you've removed zstd from your system while Emacs is running (without restarting it), or if you used the pdumper to create an image of Emacs at a time when zstd existed, and continued to use it after zstd has been removed from your system.

If neither sounds like your cause, then see if you can reproduce the issue in [vanilla Doom+](https://github.com/hlissner/doom-emacs/blob/develop/docs/getting_started.org#testing-in-dooms-sandbox).

----

If you don't have time to test it, you can get around it with

```elisp
;; add to ~/.doom.d/config.el
(setq undo-fu-session-compression nil)
```

**Comment by ymarco** [comment]
> The only way this can happen is if you've removed zstd from your system while Emacs is running (without restarting it), or if you used the pdumper to create an image of Emacs at a time when zstd existed, and continued to use it after zstd has been removed from your system.

Wouldn't this happen with leftover zstd-compressed history files as well? As in
1. Edit something with Emacs while `zstd` is present in path. Zstd-compressed history files would be saved.
2. You *can* close Emacs
2. Remove zstd from path
3. Old zstd history files are left, causing this error when next opening Emacs and editing one of the old files
I had this exact thing while moving linux distros - my new distro didn't have zstd installed, so when I copied over my .emacs.d...

**Comment by mmajewsk** [comment]
It's weird but I can't reproduce it right now, but this problem has been reocurring multiple times.
I did not experiment with the path in between the saves other than changing conda environments.
If you need to close this issues you can, but please don't delete this for good, if the problem will come up again i will try to check path for zstd and post the result here.

**Comment by hlissner** [comment]
> Wouldn't this happen with normal leftover zstd-compressed history files as well?

No. The zstd files will remain, but if zstd is absent at startup, then unfo-fu-session won't append `.zstd` to its history file paths and Emacs will not try to transparently compress/decompress them.

> I did not experiment with the path in between the saves other than changing conda environments.

This could be it. If conda is changing your `PATH` to one that doesn't include's `zstd`'s location.

**Comment by mmajewsk** [comment]
@hlissner 
> @mmajewsk This could be it. If conda is changing your `PATH` to one that doesn't include's `zstd`'s location.

Well, none of the environments did not include that, at least not installed by me explicitly.

**Comment by hlissner** [comment]
@mmajewsk Try regenerating your envvar file: run `doom env` on the command line, then restart Emacs.

**Comment by ymarco** [comment]
> @ymarco No. The zstd files will remain, but if zstd is absent at startup, then unfo-fu-session won't append .zstd to its cache file paths and Emacs will not try to transparently compress/decompress them.

I see. I might have forgotten to run `doom env` myself, back then....
